src/components/DynamicFormContainer/DynamicFormLayout.tsx

```diff
@@ -1,28 +1,110 @@
+import Ajv from "ajv";
 import React, { FunctionComponent, useState } from "react";
 import { Redirect } from "react-router";
-import { useActiveFormContext } from "../../common/context/activeForm";
 import { useConfigContext } from "../../common/context/config";
-import { SvgIcon, SvgIconArrowLeft } from "../../UI/SvgIcon";
+import { useFormsContext } from "../../common/context/forms";
+import { Button } from "../../UI/Button";
+import { SvgIcon, SvgIconArrowLeft, SvgIconTrash } from "../../UI/SvgIcon";
 import { Title } from "../../UI/Title";
 import { ToggleSwitch } from "../../UI/ToggleSwitch";
 import { Container } from "../Container";
+import { ModalDialog } from "../ModalDialog";
 import { ProgressBar } from "../ProgressBar";
 import { DynamicForm } from "./DynamicForm";
 
 export const DynamicFormLayout: FunctionComponent = () => {
   const { config } = useConfigContext();
+  const [showDeleteModal, setDeleteModal] = useState(false);
   const [isPreviewMode, setIsPreviewMode] = useState(false);
-  const { activeFormIndex, setActiveFormIndex } = useActiveFormContext();
-  if (activeFormIndex === undefined) return <Redirect to="/forms-selection" />;
-  const activeForm = config?.forms[activeFormIndex];
-  if (!activeForm) return <Redirect to="/forms-selection" />;
+  const {
+    forms,
+    setForms,
+    setActiveFormIndex,
+    currentForm,
+    setCurrentFormData,
+  } = useFormsContext();
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [formError, setFormError] = useState(false);
+  if (!currentForm) return <Redirect to="/forms-selection" />;
+  const currentFormDefinition = config?.forms[currentForm?.templateIndex];
+  if (!currentFormDefinition) return <Redirect to="/forms-selection" />;
+  if (isSubmitted) return <Redirect to="/publish" />;
 
-  const onBackToFormSelection = (): void => {
+  const formSchema = currentFormDefinition.schema;
+  const attachmentAccepted = !!currentFormDefinition.attachments?.allow;
+  const attachmentAcceptedFormat = currentFormDefinition.attachments?.accept;
+
+  const validateCurrentForm = (): boolean[] => {
+    const ajv = new Ajv();
+    const error = [] as any; // eslint-disable-line @typescript-eslint/no-explicit-any
+    forms.forEach((form) => {
+      const validForm = ajv.validate(form.data.schema, form.data.formData);
+      if (!validForm) {
+        setFormError(true);
+        console.error(ajv.errors);
+        error.push(true);
+      }
+      setFormError(false);
+      error.push(false);
+    });
+    return error;
```

Comment:
Can return true/false here instead of an array if you are not going to use that array elsewhere. 
Can make the display of the errors in this function also, just set it from inside. Then when the function first run, just unset the errors.

Reply:
ok